util/dvsim/Launcher.py

```diff
@@ -182,18 +192,13 @@ def kill(self):
 
         raise NotImplementedError()
 
-    def _has_passed(self):
+    def _check_status(self):
         """Determine the outcome of the job (P/F if it ran to completion).
 
-        Return True if the job passed, False otherwise. This is called by
+        Return a tuple of job's status and the err_msg extracted from the log.
+        Job status is "P" if the it passed, "F" otherwise. This is called by
```

Comment:
When returning tuples, it's sometimes easier to document like this: "Returns (status, err_msg), where status is ...", since it lets you combine "returns a tuple" and name the projections all at once.

(Obviously, you don't have to do this, but I've spent ages in the past trying to phrase this sort of thing and everything came out really clumsy, so I thought I'd share my "wisdom" :grin:)

Reply:
Sure, done.